Source/stores.cpp

```diff
@@ -365,7 +365,11 @@ void StartSmithBuy()
 	AddItemListBackButton();
 
 	storenumh = 0;
-	for (int i = 0; !smithitem[i].isEmpty(); i++) {
+	for (Item &item : smithitem) {
+		if (item.isEmpty())
+			continue;

```

Comment:
With all of these loops previously they would `break;` on the first empty item (which doesn't matter in practice since the arrays are sorted so there are no holes). Did you want to restore that behaviour or always run through the whole list?

Reply:
I think it's simpler to process the whole list.

The idea is to treat all arrays the same (`premiumitems` isn't sorted) and not have special requirements, like needing an empty item at the end to avoid OOB. Things should work out much the same as before since it will skip all empty items.